src/guidellm/executor/profile_generator.py

```diff
@@ -61,25 +61,24 @@ def next_profile(
         pass
 
 
-@ProfileGenerator.register_generator(ProfileGenerationModes.SINGLE)
-class SingleProfileGenerator(ProfileGenerator):
-    def __init__(self, rate: float, rate_type: str, **kwargs):
-        super().__init__(ProfileGenerationModes.SINGLE)
-        self._rate = rate
+@ProfileGenerator.register_generator(ProfileGenerationModes.MULTI)
+class MultiProfileGenerator(ProfileGenerator):
+    def __init__(self, rate: List[float], rate_type: str, **kwargs):
+        super().__init__(ProfileGenerationModes.MULTI)
+        self._rates = rate
+        self._rate_index = 0
         self._rate_type = rate_type
         self._generated = False
 
     def next_profile(
         self, current_report: TextGenerationBenchmarkReport
     ) -> Optional[Profile]:
-        if self._generated:
+        if self._rate_index >= len(self._rates):
```

Comment:
I don't believe we're incrementing the rate index anywhere, need to do that after we've grabbed the next profiles rate